documentation/modules/cruise-control/proc-implementing-rebalance-proposal.adoc

```diff
@@ -0,0 +1,78 @@
+// Module included in the following assemblies:
+//
+// assembly-cruise-control-concepts.adoc
+
+[id='proc-implementing-optimization-proposal-{context}']
+
+= Implementing an optimization proposal
+
+You can implement an xref:con-optimization-proposals-{context}[optimization proposal] if its status is `ProposalReady`.
```

Comment:
Well, I think we'll have to stick to "implement" unless there is strong agreement on the new term.

Reply:
@laidan6000 I am not a native English but tbh "implement" doesn't sound the right term here. I agree with @tombentley about "apply". Of course, I could be wrong due to the fact I am not English. @tomncooper wdyt?

Reply:
Yeah "Implement" feels wrong as it implies that the user is going to do all the steps, when actually they are just telling CC to go and do the steps. TBF "apply" suffers from the same issue. I think the problem is wrapping this in one term. It is a two step process. The user is not "Implementing an optimization proposal" they are "Obtaining and approving an optimization proposal". 

So better to say "You can _approve_ an optimization proposal if its status is `ProposalReady`."

Reply:
@ppatierno - You are definitely _not_ wrong. Please, never think that your suggestions on wording and terminology are less valid because you're a non-native English speaker.

I used "implement" to mean "carrying out" or "putting into effect" the actions summarized in the optimization proposal. It's a synonym of "execute". 

https://www.collinsdictionary.com/dictionary/english/implement

"Apply" is a more general verb, but it can also mean "to carry out". I just wanted to use a more specific verb in this case, to add variety to the documentation as much as anything else.

Reply:
I still think the issue is that the user is not "implementing/applying/executing" the optimisation proposal, they are allowing CC to do it. If we are being really picky a user could "implement" the OP by accessing the verbose OP (not yet supported) and getting the partition reassignment command json and applying it themselves.

Reply:
From one of the previous chapter ... "You can choose to apply those optimization proposals" ... so I see you used "apply"

Reply:
Well, the change above was from Tom's review. But, OK, I see your point. I'll change to "apply".

Good discussion all!

Reply:
Sorry, I re-read Tom C's comments. "Obtaining and approving an optimization proposal" might be the best choice here.

Reply:
I changed the title to "Approving an optimization proposal". All references to "implementing" proposals elsewhere in the doc have been changed to "approving".